ckanext/ontario_theme/plugin.py

```diff
@@ -422,6 +422,20 @@ def extract_package_name(url):
     get_package_name = find_package.findall(url)
     if len(get_package_name) > 0:
         return get_package_name[0]
+    else:
+        return 
+
+def extract_resource_name(url):
```

Comment:
This function is almost identical to `extract_package_name()`. Can you one generic function that works for both?

Reply:
Done!